src/swap/SwapReviewScreen.tsx

```diff
@@ -0,0 +1,374 @@
+                hitSlop={variables.iconHitslop}
+              >
+                <Text style={{ marginRight: 4, ...fontStyles.regular }}>
+                  {t('swapReviewScreen.swapFee')}
+                </Text>
+                <InfoIcon size={12} color={colors.gray4} />
+              </TouchableOpacity>
+              <TokenDisplay
+                style={[styles.transactionDetailsRightText, !swapFeeEnabled && styles.feeWaived]}
+                amount={divideByWei(swapInfo?.unvalidatedSwapTransaction?.sellAmount).multipliedBy(
+                  swapFeeDecimal
+                )}
+                tokenAddress={fromToken}
+                showLocalAmount={true}
+                testID={'SwapFee'}
+              />
+            </View>
+          </View>
+        </ScrollView>
+      )}
+      <Button
+        style={{ padding: Spacing.Regular16 }}
+        onPress={() => Logger.debug('TODO Perform Swap!!!')}
+        text={t('swapReviewScreen.complete')}
+        size={BtnSizes.FULL}
+        disabled={loading || fetchSwapQuoteError}
+      />
+      {/** Estimated amount dialog */}
+      <Dialog
+        title={t('swapReviewScreen.estimatedAmountTitle')}
+        isVisible={estimatedModalVisible}
+        actionText={t('dismiss')}
+        actionPress={() => setEstimatedDialogVisible(false)}
+        isActionHighlighted={false}
+        onBackgroundPress={() => setEstimatedDialogVisible(false)}
+      >
+        {t('swapReviewScreen.estimatedAmountBody', {
+          slippagePercent: maxSlippagePercent,
+        })}
+      </Dialog>
+      {/** App fee dialog */}
+      <Dialog
+        title={t('swapReviewScreen.swapFeeTitle')}
+        isVisible={swapFeeModalVisible}
+        actionText={t('dismiss')}
+        actionPress={() => setSwapFeeModalVisible(false)}
+        isActionHighlighted={false}
+        onBackgroundPress={() => setSwapFeeModalVisible(false)}
+      >
+        {t('swapReviewScreen.swapFeeBody', {
+          swapFee: swapFeePercentage,
+        })}
+      </Dialog>
+    </SafeAreaView>
+  )
+}
+
+const styles = StyleSheet.create({
+  safeAreaContainer: {
+    flex: 1,
+  },
+  loadingContentContainer: {
+    justifyContent: 'flex-start',
+    alignItems: 'center',
+    padding: Spacing.Regular16,
+    flexGrow: 1,
+  },
+  contentContainer: {
+    padding: Spacing.Regular16,
+    flexGrow: 1,
+  },
+  subContentContainer: {
+    paddingBottom: Spacing.Regular16,
+  },
+  tallRow: {
+    flexDirection: 'row',
+    justifyContent: 'space-between',
+    paddingBottom: Spacing.Regular16,
+  },
+  row: {
+    flexDirection: 'row',
+    justifyContent: 'space-between',
+    paddingBottom: Spacing.Smallest8,
+  },
+  tokenDisplayView: {
+    flex: 1,
+    justifyContent: 'space-between',
+  },
+  separator: {
+    height: 1,
+    width: '100%',
+    backgroundColor: colors.gray2,
+  },
+  transactionDetailsRightText: {
+    ...fontStyles.regular,
+    flex: 1,
+    alignSelf: 'flex-end',
```

Comment:
i think `alignSelf` and `textAlignVertical` are not needed here?

Reply:
Fixed in https://github.com/valora-inc/wallet/pull/2767/commits/7d463246666579c3467bbcc2e82dff8142660a72!